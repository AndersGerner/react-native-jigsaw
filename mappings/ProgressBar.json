[{"name":"Progress Bar","tag":"ProgressBar","description":"A horizontal bar used to show completed progress","category":"formControl","preview_image_url":"https://res.cloudinary.com/altos/image/upload/draftbit/library/jigsaw-1.0/reps/Status_Progress.png","supports_list_render":false,"props":{"progress":{"label":"Progress","description":"The amount of progress to display. A number 0-1.","type":"number","value":0.5,"min":0,"max":1,"step":0.01,"precision":2,"editable":true,"required":true},"color":{"label":"Progress Color","description":"Custom color for the progress shown","type":"color","value":null,"editable":true,"required":true},"unfilledColor":{"label":"Unfilled Color","description":"The color of the unfilled portion of the progress bar(eg. if at 50% then this is the color of the other 50%)","type":"color","value":null,"editable":true,"required":true},"borderRadius":{"label":"Border Radius","description":"The border radius of the bar","type":"number","value":10,"min":0,"max":100,"step":1,"precision":1,"editable":true,"required":true},"borderWidth":{"label":"Border Width","description":"The width of the border that surrounds the bar.","type":"number","value":1,"min":0,"max":15,"step":1,"precision":1,"editable":true,"required":true},"borderColor":{"label":"Border Color","description":"Custom color for border of the entire bar","type":"color","value":null,"editable":true,"required":true},"animationType":{"label":"Animation Type","description":"The type of animation that occurs when the bar is filled(Default is Spring)","type":"flatArray","value":"spring","options":["decay","timing","spring"],"editable":true,"required":true}},"layout":{"width":200,"height":20}}]