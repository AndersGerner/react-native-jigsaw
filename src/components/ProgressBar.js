import * as React from "react"
import * as Progress from "react-native-progress"
import { COMPONENT_TYPES, FORM_TYPES } from "../core/component-types"
import { withTheme } from "../../core/theming"

const ProgressBar = ({
  progress,
  style,
  borderRadius,
  animationType,
  progressColor,
  unfilledColor,
  borderColor,
  borderWidth
}) => {
  return (
    <Progress.Bar
      progress={progress}
      color={progressColor}
      unfilledColor={unfilledColor}
      borderColor={borderColor}
      borderWidth={borderWidth}
      width={style.width}
      height={style.height}
      borderRadius={borderRadius}
      animationType={animationType}
    />
  )
}

export default withTheme(ProgressBar)

export const SEED_DATA = [
  {
    name: "Progress Bar",
    tag: "ProgressBar",
    description: "A horizontal bar used to show completed progress",
    category: COMPONENT_TYPES.formControl,
    preview_image_url: "{CLOUDINARY_URL}/Status_Progress.png",
    supports_list_render: false,
    props: {
      progress: {
        label: "Progress",
        description: "The amount of progress to display. A number 0-1.",
        type: FORM_TYPES.number,
        value: 0.5,
        min: 0,
        max: 1,
        step: 0.01,
        precision: 2,
        editable: true
      },
      progressColor: {
        label: "Progress Color",
        description: "Custom color for the progress shown",
        editable: true,
        value: null,
        required: true,
        type: FORM_TYPES.color
      },
      unfilledColor: {
        label: "Unfilled Color",
        description:
          "The color of the unfilled portion of the progress bar(eg. if at 50% then this is the color of the other 50%)",
        editable: true,
        value: null,
        required: true,
        type: FORM_TYPES.color
      },
      borderRadius: {
        label: "Border Radius",
        description: "The border radius of the bar",
        type: FORM_TYPES.number,
        value: 10,
        min: 0,
        max: 100,
        step: 1,
        precision: 1,
        editable: true,
        required: true
      },
      borderWidth: {
        label: "Border Width",
        description: "The width of the border that surrounds the bar.",
        type: FORM_TYPES.number,
        value: 1,
        min: 0,
        max: 15,
        step: 1,
        precision: 1,
        editable: true,
        required: true
      },
      borderColor: {
        label: "Border Color",
        description: "Custom color for border of the entire bar",
        editable: true,
        value: null,
        required: true,
        type: FORM_TYPES.color
      },
      animationType: {
        label: "Animation Type",
        description: "The type of animation that occurs when the bar is filled(Default is Spring)",
        type: FORM_TYPES.flatArray,
        value: "spring",
        options: ["decay", "timing", "spring"],
        editable: true,
        required: true
      }
    },
    layout: {
      width: 343,
      height: 2
    }
  }
]
