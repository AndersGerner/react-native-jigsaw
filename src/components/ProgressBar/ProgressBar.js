/* @flow */

import * as React from "react";
import ProgressBarComponent from "./ProgressBarComponent";
import { withTheme } from "../../core/theming";
import type { Theme } from "../../types";

type Props = {|
  /**
   * Progress value (between 0 and 1).
   */
  progress: number,
  style?: any,
  /**
   * @optional
   */
  theme: Theme
|};

/**
 * Progress bar is an indicator used to present progress of some activity in the app.
 *
 * ## Usage
 * ```js
 * import * as React from 'react';
 * import { ProgressBar } from '@draftbit/ui';
 *
 * const MyComponent = () => (
 *   <ProgressBar progress={0.5} />
 * );
 *
 * export default MyComponent;
 * ```
 */
class ProgressBar extends React.Component<Props> {
  render() {
    const {
      progress,
      style,
      theme: { colors }
    } = this.props;

    return (
      <ProgressBarComponent
        progress={progress}
        progressTintColor={colors.primary}
        style={style}
        trackTintColor={colors.divider}
      />
    );
  }
}

export default withTheme(ProgressBar);
