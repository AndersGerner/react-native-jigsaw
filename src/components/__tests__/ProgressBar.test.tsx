import * as React from "react";
import renderer from "react-test-renderer";
import ProgressBar from "../ProgressBar";

it("renders progress bar", () => {
  const tree = renderer.create(<ProgressBar />).toJSON();

  expect(tree).toMatchSnapshot();
});
