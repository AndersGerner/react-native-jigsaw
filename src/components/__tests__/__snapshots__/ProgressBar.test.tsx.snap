// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders progress bar 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": undefined,
        "borderColor": "rgba(0, 122, 255, 1)",
        "borderRadius": 4,
        "borderWidth": 1,
        "overflow": "hidden",
        "width": 150,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(0, 122, 255, 1)",
        "height": 6,
        "transform": Array [
          Object {
            "translateX": 0,
          },
          Object {
            "translateX": -74,
          },
          Object {
            "scaleX": 0.0001,
          },
        ],
      }
    }
  />
</View>
`;
